PubFactory Journals.js

```diff
@@ -0,0 +1,458 @@
+{
+	"translatorID": "8d1fb775-df6d-4069-8830-1dfe8e8387dd",
+	"label": "PubFactory Journals",
+	"creator": "Brendan O'Connell",
+	"target": "^https?:\\/\\/(?:[\\w-]+\\.)+[\\w-]+(?:\\/[^\\/\\s]*)+\\.xml(?:\\?.*)?$|^https:\\/\\/(?:[\\w-]+\\.)+[\\w-]+\\/search\\?[^\\/\\s]*source=%2Fjournals%2F\\w+%2F\\w+-overview\\.xml(?:&q=[^&\\s]*)?$",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-03-29 13:58:05"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2023 Brendan O'Connell
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc) {
+	// EM works well on single articles, so detectWeb() isn't strictly necessary for them.
+	// However, single articles are included in this translator
+	// so that 'multiple' doesn't get called for articles with "Related Items"
+	// e.g. https://avmajournals.avma.org/view/journals/javma/261/4/javma.22.11.0518.xml
+	// because there are multiple items that match the rows selector in getSearchResults and
+	// haven't found a way to write a querySelectorAll() that only matches the main article on single pages
+	// with related items.
+	if (doc.querySelector('meta[name="citation_title"]')) {
+		return "journalArticle";
+	}
+	else if (getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('a.c-Button--link[href*="/view/journals/"][href$=".xml"][target="_self"]');
```

Comment:
Thanks for the review! I'm working my way through this. While the selector you suggest works fine for https://journals.ametsoc.org/view/journals/amsm/59/1/amsm.59.issue-1.xml, it doesn't work for https://journals.humankinetics.com/view/journals/jab/jab-overview.xml. When I was working on this translator, I discovered that PubFactory journals are pretty tricky because unlike a lot of journal platforms, the page structure differs very significantly between journals, so finding selectors that work across all of them is challenging! You're correct that my current selector does return some false positives; I'll see if I can improve it a bit. Also open to suggestions for simpler selectors! In the Test Cases, you can see some of the different journals this needs to work across.

Reply:
Thanks, I think this is a great solution. Fortunately, we don't have to contend with 1300 journals on different websites - based on their client list https://www.pubfactory.com/our-clients/, there's 23 or so publisher websites. I added a few more to the Test Cases, and they all work. For the larger publishers, e.g. Oxford University Press, they have a huge catalog of journals across a variety of platforms, and it's very hard to find specific ones that use PubFactory. So I agree with you, it would be very hard to determine if this solution truly works across the board, but it works on everything that I could test.